models/vgg/vgg.hpp

```diff
@@ -0,0 +1,181 @@
+/**
+ * @file vgg.hpp
+ * @author Shubham Agrawal
+ * 
+ * Definition of VGG models.
+ * 
+ * For more information, kindly refer to the following paper.
+ * 
+ * Paper for VGG.
+ *
+ * @code
+ * @article{Simonyan2014,
+ *  author = {Karen Simonyan, Andrew Zisserman},
+ *  title = {Very Deep Convolutional Networks for Large-Scale Image Recognition},
+ *  year = {2014},
+ *  url = {https://arxiv.org/pdf/1409.1556.pdf}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MODELS_MODELS_VGG_VGG_HPP
+#define MODELS_MODELS_VGG_VGG_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+#include <mlpack/methods/ann/loss_functions/binary_cross_entropy_loss.hpp>
+
+namespace mlpack {
+namespace models {
+
+/**
+ * Definition of a VGG CNN.
+ * 
+ * NOTE: Note that output size will be 1x1xN. Here, N is number of classes.
+ * 
+ * @tparam MatType Matrix representation to accept as input and use for
+ *    computation.
+ * @tparam VGGVersion Version of VGG.
+ * @tparam IsBatchNorm Whether to apply Batch Norm layer.
+ */
+template<
+  typename MatType = arma::mat,
+  size_t VGGVersion = 11,
+  bool IsBatchNorm = false
+>
+class VGGType : public ann::MultiLayer<MatType>
+{
+ public:
+  //! Create the VGGType layer.
+  VGGType();
+
+  /**
+   * VGGType constructor intializes number of classes and weights.
+   *
+   * @param numClasses Optional number of classes to classify images into,
+   *     only to be specified if includeTop is  true.
```

Comment:
I don't think that I understand what `includeTop` does; can you clarify that in the documentation?